### ISSUE TITLE
Binance POST /sapi/v1/sub-account/universalTransfer signature issue

### ISSUE THREAD

**Issue by geknow** [ccxt/ccxt#19491]
### Operating System

linux

### Programming Languages

Python

### CCXT Version

1.57.10

### Description

```params = {
    'fromEmail': 'xxxxxx+16@gmail.com',
    'toEmail': 'xxxxx@gmail.com',
    'futuresType': 1,
    'asset': 'USDT',
    'amount': 10,
    'recvWindow': 10000,
    
}
transfer = exchange.sapi_post_sub_account_futures_internaltransfer(params=params)
```

I got this error `ccxt.base.errors.AuthenticationError: binance {"code":-1022,"msg":"Signature for this request is not valid."}`
but when I change the email to other email with the character `+`, It works. And I changed other emails and found `+` is a specail charater.

So how to solve this?

### Code

```
  

```

**Comment by geknow** [comment]
And I also upgrade to `4.1.8` version, but it didn't work.

**Comment by geknow** [comment]
===>
but when I change the email to other emails `without` the character `+`, It works. And I changed other emails and found `+` is a special character.

**Comment by carlosmiei** [comment]
Hello @geknow just in case, can you try to replace the '+' with  `%2B` ?

Example:

```
    'fromEmail': 'xxxxxx%2B16@gmail.com',
    'toEmail': 'xxxxx@gmail.com',
```

**Comment by geknow** [comment]
> Hello @geknow just in case, can you try to replace the '+' with `%2B` ?
> 
> Example:
> 
> ```
>     'fromEmail': 'xxxxxx%2B16@gmail.com',
>     'toEmail': 'xxxxx@gmail.com',
> ```

Yes, I tried. But it didn't work. Not only the `+`, I also try `@`

**Comment by geknow** [comment]
> Hello @geknow just in case, can you try to replace the '+' with `%2B` ?
> 
> Example:
> 
> ```
>     'fromEmail': 'xxxxxx%2B16@gmail.com',
>     'toEmail': 'xxxxx@gmail.com',
> ```

I tried this email `xxxxxuan%2B16@gmail.com`

**Comment by kroitor** [comment]
Hello @geknow, closing this during a cleanup — the working answer stands in the thread from back then: a plus sign inside an email parameter must be percent encoded as %2B when passed to the signed sapi endpoints, since a raw plus decodes as a space and breaks the signature, and the report was filed against a very old version of the library besides. If the encoding still trips the signature on a current version even with %2B, a fresh issue with a verbose log would be very welcome. Thanks for the report back then!
